godef_test.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"go/build"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/rogpeppe/godef/go/types"
+	"golang.org/x/tools/go/packages/packagestest"
+)
+
+func TestGoDef(t *testing.T) { packagestest.TestAll(t, testGoDef) }
+func testGoDef(t *testing.T, exporter packagestest.Exporter) {
+	const godefAction = ">"
+	modules := []packagestest.Module{{
+		Name: "github.com/rogpeppe/godef",
+		Files: map[string]interface{}{
+			"a/a.go":      packagestest.Copy("testdata/a/a.go"),
+			"a/random.go": packagestest.Copy("testdata/a/random.go"),
+			"b/b.go":      packagestest.Copy("testdata/b/b.go"),
+		},
+	}}
+	exported := packagestest.Export(t, exporter, modules)
+	defer exported.Cleanup()
+
+	posStr := func(p token.Position) string {
+		return localPos(p, exported, modules)
+	}
+
+	const gopathPrefix = "GOPATH="
+	const gorootPrefix = "GOROOT="
+	for _, v := range exported.Config.Env {
+		if strings.HasPrefix(v, gopathPrefix) {
+			build.Default.GOPATH = v[len(gopathPrefix):]
+		}
+		if strings.HasPrefix(v, gorootPrefix) {
+			build.Default.GOROOT = v[len(gorootPrefix):]
+		}
+	}
+
+	count := 0
+	exported.Expect(map[string]interface{}{
+		"godef": func(src, target token.Position) {
+			count++
+			input, err := ioutil.ReadFile(src.Filename)
```

Comment:
Please include the tests that check that the edit buffer is correctly used (in the other PR, this involved checking to see if there was a ".saved" equivalent file and copying it over the source file for the duration of the fundtion).

Reply:
Done